packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerToolbar.tsx

```diff
@@ -56,6 +56,15 @@ const DateRangePickerToolbarContainer = styled('div', {
   display: 'flex',
 });
 
+/**
+ * Demos:
+ *
+ * - [DateRangePicker](https://mui.com/x/react-date-pickers/date-range-picker/)
```

Comment:
WDYT about adding `DateRangePickerToolbar` to the `components` list on the `date-range-picker.md` to make this correct and tie the actual generated list in API page with this comment?
<img width="1301" alt="Screenshot 2023-10-12 at 11 31 07" src="https://github.com/mui/mui-x/assets/4941090/78806b54-90db-4263-b593-22cbca323221">

Reply:
We could also add it to the `custom-components.md` `components` list as it also makes sense and would allow syncing this list with the one in the `API` page.
```suggestion
 * - [DateRangePicker](https://mui.com/x/react-date-pickers/date-range-picker/)
 * - [Custom components](https://mui.com/x/react-date-pickers/custom-components/)
```

Reply:
WDYT about https://github.com/mui/mui-x/pull/10646/files#r1356476865?
Would you agree that it makes sense?
If it does, would you make it a separate PR?

Reply:
mmh, I think this makes sense to be handled in a separate PR tbh. But first I'd like to create an issue to track any unreferenced components or those that could benefit from a (cross-)reference.

WDYT @LukasTy ?

Reply:
Separating those makes sense, although, we might change some of these comments afterward. 🙈 
That's the "beauty" of manual changes. 🤯